backend/contracts/OasisReward.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/utils/Base64.sol";
+
+contract OasisReward is ERC721Enumerable {
+    // Mapping from token ID to IPFS URI
```

Comment:
Please use the [natspec](https://docs.soliditylang.org/en/latest/natspec-format.html) format where possible. Same for other variables.

```suggestion
    /// Mapping from token ID to IPFS URI.
```

Reply:
Ok, added natspec format to OasisReward.sol.